src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -352,18 +347,23 @@ def resample(self, timestamp: datetime) -> Sample:
                 If there are no *relevant* samples, then the new sample will
                 have `None` as `value`.
         """
-        threshold = timestamp - timedelta(
-            seconds=self._config.max_data_age_in_periods
-            * self._config.resampling_period_s
+        conf = self._config
+        minimum_relevant_timestamp = timestamp - timedelta(
+            seconds=conf.resampling_period_s * conf.max_data_age_in_periods
         )
-        self._remove_outdated_samples(threshold=threshold)
-
+        # We need to pass a dummy Sample to bisect because it only support
+        # specifying a key extraction function in Python 3.10, so we need to
+        # compare samples at the moment.
+        cut_index = bisect(self._buffer, Sample(minimum_relevant_timestamp, None))
```

Comment:
Sorry, I am outdated...
Why we need bisect in the ring buffer?

Reply:
Because now it is fixed size, so we add samples as they come and never removed them, only when they "drop out" of the circular buffer. If we stop receiving for example, the buffer will be filled, but with old data, so we need to make sure to pass only relevant data to the resampling function. I'm bisecting to find the oldest relevant timestamp in the buffer.